router/pkg/httputil/headers.go

```diff
@@ -0,0 +1,27 @@
+package httputil
```

Comment:
I think we're using very similar code somewhere else in the Router.
Can we consolidate this and go away from calling a package "httputil" (not idiomatic go) and instead give files meaning?